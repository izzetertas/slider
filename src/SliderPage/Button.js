import React from 'react'

const Button = ({ label, onClick, disabled }) => (
  <input
    type='button'
    onClick={onClick}
    value={label}
    disabled={disabled}
    className='button'
  />
)

export default Button;
