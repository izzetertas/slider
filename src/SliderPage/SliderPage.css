@import "~slick-carousel/slick/slick.css";
@import "~slick-carousel/slick/slick-theme.css";

.container {
  margin: 0 auto;
  padding: 40px;
  width: 90%;
  color: #333;
}

.input-container {
  display: flex;
}

input{
  height: 40px;
  font-size: 18px;
}

input[type="text"] {
  width: 100%;
}

button {
  width: 140px;
  margin-left: 15px;
}

h3 {
  border: 1px solid;
  font-size: 36px;
  line-height: 100px;
  margin: 10px;
  padding: 2%;
  position: relative;
  text-align: center;
}
.variable-width .slick-slide p {
  background: blue;
  height: 100px;
  color: #fff;
  margin: 5px;
  line-height: 100px;
  text-align: center;
}
.center .slick-center h3 {
  color: #e67e22;
  opacity: 1;
  transform: scale(1.08);
}
.center h3 {
  opacity: 0.8;
  transition: all 300ms ease;
}
.content {
  padding: 20px;
  margin: auto;
  width: 90%;
}
.slick-slide .image {
  padding: 10px;
}
.slick-slide img {
  border: 5px solid #fff;
  display: block;
  margin: auto;
}
.slick-slide img.slick-loading {
  border: 0;
}
.slick-slider {
  margin: 30px auto 50px;
}
.slick-dots {
  margin-left: 0;
}
.slick-thumb {
  bottom: -45px;
}
.slick-thumb li {
  width: 60px;
  height: 45px;
}
.slick-thumb li img {
  filter: grayscale(100%);
}
.slick-thumb li.slick-active img {
  filter: grayscale(0%);
}

@media (max-width: 768px) {
  h3 {
    font-size: 24px;
  }
  .center {
    margin-left: -40px;
    margin-right: -40px;
  }
  .center .slick-center h3 {
    color: #e67e22;
    opacity: 1;
    transform: scale(1);
  }
  .center h3 {
    opacity: 0.8;
    transform: scale(0.95);
    transition: all 300ms ease;
  }
}
.slick-vertical .slick-slide {
  height: 180px;
}
/* .slick-arrow {
  background-color: grey;
  color: black;
}
.slick-arrow:hover {
  background-color: grey;
} */
/* .slick-prev {
  z-index: 100;
  left: 20px !important;
}
.slick-next {
  z-index: 100;
  right: 20px !important;
} */

.slick-prev,
.slick-next { 
  top: 35%;
}

.slick-prev:before,
.slick-next:before { 
  font-family: "slick";
  font-size: 40px;
  line-height: 1;
  color: black;
  opacity: 0.75;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}   

.slick-prev:before {
  content: "‹";
}

[dir="rtl"]
.slick-prev:before {
  content: "›";
}

[dir="rtl"]
.slick-next {
  left: -10px;
  top: 50px;
  right: auto;
}

.slick-next:before {
  content: "›";
}

[dir="rtl"]
.slick-next:before {
  content: "‹";
}