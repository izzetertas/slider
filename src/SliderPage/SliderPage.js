import React, { Component } from 'react'
import Slider from 'react-slick'
import './SliderPage.css'

class SliderPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [],
      title: ''
    }
  }
  
  componentDidMount(){
    this.focusToInput()
  }

  focusToInput = () => {
    this.textInput && this.textInput.focus()
  }

  handleTitleOnChange = (e) =>  this.setState({ title : e.target.value })

  handleTitleOnKeyPress = (e) =>  e.key === 'Enter' && this.handleOnClick()

  handleOnClick = (e) => {
    const { items, title } = this.state
    if(!title) return

    this.setState({
      items: [...items, title],
      title: ''
    })
    this.focusToInput()
  }

  render(){
    const { items, title } = this.state;
    const settings = {
      dots: false,
      infinite: false,
      speed: 500,
      slidesToShow: 3,
      slidesToScroll: 1,
      responsive: [
        {
          breakpoint: 600,
          settings: {
            dots: true
          }
        }
      ]
    };

    return (
      <div className='container'>
        <div className='input-container'>
          <input
            type='text'
            placeholder='Enter a title'
            autoFocus
            value={title}
            className='title'
            ref={input => this.textInput = input} 
            onChange={this.handleTitleOnChange}
            onKeyPress={this.handleTitleOnKeyPress}
          />
          <button
            onClick={this.handleOnClick}>
            Add New Slide
          </button>
        </div>
        <div>
          <Slider {...settings}>
            {items &&
              items.map((item, i) => (
                <div className='item' key={i}>
                  <h3>{item}</h3>
                </div>
              ))
            }
          </Slider>
        </div>
      </div>
    );
  }
}

export default SliderPage;