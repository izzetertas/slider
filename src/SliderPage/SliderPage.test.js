import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import SliderPage from './SliderPage'

let initialState;
let wrapper;

describe('<SliderPage/>', () => {
  beforeEach(() => {
    initialState = {
      items: []
    }
    wrapper = shallow(<SliderPage />)
  })

  it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      // expect(wrapper.find('button').length).toBe(1)
      // expect(wrapper.find('input').find({type: 'text'}).length).toBe(1)
      // expect(wrapper.find('input').find({type: 'text'}).text()).toEqual('')
  });

  it.skip('should render with items correctly', () => {
    wrapper.setState({ items: ['Test 1', 'Test 2', ' Test 3', 'Test 4']})
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it.skip('should set initial state correctly', () => {
  //   expect(wrapper.state()).toEqual(initialState)
  // });

  // describe('when user clicks Add New Slider button', () => {
  //   it.skip('should not change state if title is empty', () => {
  //     wrapper.setState({ title: '' })
  //     wrapper.find('button').first().simulate('click')
  //     expect(wrapper.state('items')).toEqual([])
  //   });

  //   it.skip('should add text value to items of the state', () => {
  //     wrapper.setState({ title: '1' })
  //     wrapper.find('button').first().simulate('click')
  //     const expected = ['1']
  //     expect(wrapper.state('items')).toEqual(expected)
  //   });
  // });


  // it.skip('should call textInput when input focuses', () => {
  //   wrapper = mount(<SliderPage />)
  //   const { textInput } = wrapper.instance();
  //   jest.spyOn(textInput, "focus");
  //   wrapper.instance().componentDidMount();
  //   expect(textInput.focus).toHaveBeenCalledTimes(1);
  // })

  // describe('when user type in text input', () => {
  //   it.skip('should update state correctly', () => {
  //     wrapper.find('input').find({type: 'text'}).first().simulate('change',  { target: { value: 'Test 1' } })
  //     expect(wrapper.state('title')).toEqual('Test 1')
  //   });
  // });

  // describe('when user click enter in text input', () => {
  //   it.skip('should call button click method', () => {
  //     wrapper.setState({ items: ['Test 1'], title: 'Test 2' })
  //     wrapper.find('input').find({type: 'text'}).first().simulate('keyPress',  { key: 'Enter' })
  //     expect(wrapper.state('items')).toEqual(['Test 1', 'Test 2'])
  //     expect(wrapper.state('title')).toEqual('')
  //   });
  // });  
});