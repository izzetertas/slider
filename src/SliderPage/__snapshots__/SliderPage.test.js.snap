// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderPage/> should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="input-container"
  >
    <input
      autoFocus={true}
      className="title"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter a title"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add New Slide
    </button>
  </div>
  <div>
    <Slider
      dots={false}
      infinite={false}
      responsive={
        Array [
          Object {
            "breakpoint": 600,
            "settings": Object {
              "dots": true,
            },
          },
        ]
      }
      slidesToScroll={1}
      slidesToShow={3}
      speed={500}
    />
  </div>
</div>
`;

exports[`<SliderPage/> should render with items correctly 1`] = `
<div
  className="container"
>
  <div
    className="input-container"
  >
    <input
      autoFocus={true}
      className="title"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter a title"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add New Slide
    </button>
  </div>
  <div>
    <Slider
      dots={false}
      infinite={false}
      responsive={
        Array [
          Object {
            "breakpoint": 600,
            "settings": Object {
              "dots": true,
            },
          },
        ]
      }
      slidesToScroll={1}
      slidesToShow={3}
      speed={500}
    >
      <div
        className="item"
        key="0"
      >
        <h3>
          Test 1
        </h3>
      </div>
      <div
        className="item"
        key="1"
      >
        <h3>
          Test 2
        </h3>
      </div>
      <div
        className="item"
        key="2"
      >
        <h3>
           Test 3
        </h3>
      </div>
      <div
        className="item"
        key="3"
      >
        <h3>
          Test 4
        </h3>
      </div>
    </Slider>
  </div>
</div>
`;
